lighthouse-core/gather/gather-runner.js

```diff
@@ -86,13 +86,20 @@ class GatherRunner {
    * @return {!Promise}
    */
   static loadPage(driver, options) {
-    return driver.gotoURL(options.url, {
-      waitForLoad: true,
-      disableJavaScript: !!options.disableJavaScript,
-      flags: options.flags,
-    }).then(finalUrl => {
-      options.url = finalUrl;
-    });
+    const resetStorage = !options.flags.disableStorageReset;
+    const recordTrace = options.config.recordTrace;
+    const useThrottling = options.config.useThrottling;
+    return Promise.resolve()
+      // Clear disk & memory cache if it's a perf run
+      .then(_ => resetStorage && recordTrace && useThrottling && driver.cleanBrowserCaches())
```

Comment:
this heuristic seems a little magical, but I can't think of a better one. At some point we'll want an escape hatch (for if you want a perf pass but not a trace...e.g. running your own gatherers or in-page metrics), but in theory finishing #1826 would mean you'd be able to set a `disableStorageReset` per-pass in the config